File: manual/code/test_suites/python/python/test_levenshtein_hamming.py
import unittest
from function_levenshtein_hamming import StringProcessor

class TestHammingDistance(unittest.TestCase):

    def setUp(self):
        self.processor = StringProcessor()

    def test_equal_strings(self):
        result = self.processor.calculate_hamming_distance("test", "test")
        self.assertEqual(result, 0)

    def test_single_character_difference(self):
        result = self.processor.calculate_hamming_distance("objective", "objection")
        self.assertEqual(result, 2)

    def test_insertions_and_deletions(self):
        result = self.processor.calculate_hamming_distance("hello", "helo")
        self.assertEqual(result, "Strings must be of the same length for Hamming distance.")


    def test_single_substitution(self):
        result = self.processor.calculate_hamming_distance("flaw", "lawn")
        self.assertEqual(result, 4)  

    def test_non_string_input(self):
        result = self.processor.calculate_hamming_distance(123, "hello")
        self.assertEqual(result,"Error: Both inputs must be strings")

if __name__ == "__main__":
    unittest.main()

File: manual/data/java/tests/test_levenshtein_hamming.java
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class test_levenshtein_hamming {
    private function_levenshtein_hamming processor;

    @BeforeEach
    void setUp() {
        processor = new function_levenshtein_hamming();
    }

    @Test
    void test_equal_strings() {
        String result = processor.calculate_hamming_distance("test", "test");
        assertEquals("0", result);
    }

    @Test
    void test_single_character_difference() {
        String result = processor.calculate_hamming_distance("objective", "objection");
        assertEquals("2", result);
    }

    @Test
    void test_insertions_and_deletions() {
        String result = processor.calculate_hamming_distance("hello", "helo");
        assertEquals("Strings must be of the same length for Hamming distance.", result);
    }

    @Test
    void test_single_substitution() {
        String result = processor.calculate_hamming_distance("flaw", "lawn");
        assertEquals("4", result);
    }

    @Test
    void test_non_string_input() {
        String result = processor.calculate_hamming_distance(null, "hello");
        assertEquals("Error: Both inputs must be strings", result);
    }
}